### DISCUSSION TITLE
Pandas occupies memory when slicing even if no variable is created

### DISCUSSION THREAD

**Discussion by jaymc-betta** [microsoft/vscode-jupyter#14190] [Questions and Answers]
I'm slicing a dataset with `loc` `iloc` `filter` and not saving it to a varaible because I'm exploring a dataset, but it occupies RAM. How can I avoid this? Is it normal? I've never had this issue before, and I work a lot with **jupyter in vscode**. What's happening?

```python

df.iloc[0:100000] # increases RAM every time I execute it!

```

I've tried this same code in ipython terminal, and does not increases my RAM usage. So this must be a thing of jupyter+vscode. Any clue?

Edit: I'm using .py files for this (not ipynb) with text selection execution configuration, but I allways do...

  [1]: https://stackoverflow.com/questions/76984731/ram-usage-rises-after-printing-dataframe-copy-in-pandas/76984775#76984775

**Comment by DonJayamanne** [comment]
Closing discussion in favour of a duplicate issue https://github.com/microsoft/vscode-jupyter/issues/14189
